src/server/types.ts

```diff
@@ -16,5 +16,6 @@ declare namespace ts.server {
         gc?(): void;
         trace?(s: string): void;
         require?(initialPath: string, moduleName: string): RequireResult;
+        importServicePlugin?(root: string, moduleName: string): Promise<any>;

```

Comment:
since `importServicePlugin` is serving a similar purpose as `require`, we should probably have the signatures match:

```suggestion
        importAsync?(initialPath: string, moduleName: string): Promise<RequireResult>;
```

In general we've tried to avoid `try..catch` in the compiler and language server (with the exception of cancellation), which is why we return `RequireResult` for `require` (which returns an object with an `error` property). This would also help make handling the result more consistent across both `require` and this method. 

I also recommend changing the name to be less specific to service plugins in case we want to leverage it for other purposes in the future.

Reply:
The current implementation is specific to plugins which simplifies the code instead of having to implement a generic `importAsync`

I'll look into aligning using `RequireResult` here but do not want to cause confusion by making the method seem more generic than it actually is intended to be

Reply:
> The current implementation is specific to plugins which simplifies the code instead of having to implement a generic `importAsync`

That's a valid reason. We can leave the name as-is then.